packages/react-storage/src/components/StorageBrowser/do-not-import-from-here/actions/listLocationItemsAction.ts

```diff
@@ -99,6 +133,17 @@ export async function listLocationItemsAction(
     },
   };
 
+  if (prefetch) {
+    return await listWithPrefetch(
+      path,
+      {
+        ...listInput.options,
+        pageSize: MAX_PAGE_SIZE,
+      },
+      prefetch
+    );
+  }
```

Comment:
Looks like a consumer might find out `prefetch` handling somewhat non-deterministic if they provide a `pageSize` and/or `nextToken` along with the `prefetch` flag?